tlatools/org.lamport.tlatools/src/tlc2/tool/Simulator.java

```diff
@@ -399,9 +401,8 @@ protected int simulate(final StateVec initStates) throws InterruptedException {
 				// see tlc2.tool.Worker.doPostCheckAssumption()
 				if (result.error().hasTrace()) {
 					error.errorCode = Math
-							//TODO numOfGenTraces is not ideal because it is not monotonically increasing.
 							.max(this.tool.checkPostConditionWithCounterExample(result.error().getCounterExample(),
-									Map.of("_DumpTraceFilePrefix", String.format("%s_", numOfGenTraces.get()))), error.errorCode);
+									Map.of("_DumpTraceFilePrefix", String.format("%s_", System.currentTimeMillis()))), error.errorCode);
```

Comment:
The `currentTimeMillis` causes the file names to be unpredictable. While I don’t necessarily suggest that it should be `numOfGenTraces.get`, `numOfGenTraces.get` does have the advantage of allowing you to deduce the name of the dump file from `TLCGet(“stats”)`.

Reply:
OK. I'll return previous implementation back.